engine/include/cubos/engine/renderer/renderer.hpp

```diff
@@ -38,6 +38,14 @@ namespace cubos::engine
     /// @ingroup renderer-plugin
     using Renderer = std::shared_ptr<BaseRenderer>;
 
+    /// @brief Struct which holds the viewport information for a camera, to be used for drawing.
+    /// @ingroup renderer-plugin
+    struct Viewport
+    {
+        glm::ivec2 position;
+        glm::ivec2 size;
+    };
```

Comment:
I think we should move this type to the public section of the `BaseRenderer` class, since it's only used in renderer code.